lib/search_vector.rb

```diff
@@ -5,15 +5,32 @@ module SearchVector
   AcceptedTables = ['supporters', 'payments']
 
   def self.query(query_string, expr=nil)
-    query = if query_string.is_a?(Integer) || query_string.is_int?
+    query = if (query_string.is_a?(Integer) || query_string.is_int?) && SearchVector.within_postgres_integer_limit(query_string)
       "(supporters.fts @@ websearch_to_tsquery('english', $search::varchar(255))
         OR donations.fts  @@ websearch_to_tsquery('english', $search::varchar(255))
+        OR (
+          supporters.phone IS NOT NULL
+          AND supporters.phone != ''
+          AND supporters.phone_index = $search::varchar(255)
+        )
         OR payments.id = $search::INTEGER)"
     else
       "(supporters.fts @@ websearch_to_tsquery('english', $search)
+        OR (
+          supporters.phone IS NOT NULL
+          AND supporters.phone != ''
+          AND supporters.phone_index IS NOT NULL 
+          AND supporters.phone_index != ''
+          AND supporters.phone_index = (regexp_replace($search, '\\D','', 'g'))
```

Comment:
Is it right? I thought it would be with a single `\` instead.

```suggestion
          AND supporters.phone_index = (regexp_replace($search, '\D','', 'g'))
```

Reply:
It's correct with two `\`. There's some sort of escaping happening in Postgres or Ruby that strips the first one `\`. I tried with a single `\` and it doesn't work.

Reply:
I actually tried it again with a single `\` and it does fail.

Reply:
Ahh, I forgot about that!